packages/flutter/lib/src/material/carousel.dart

```diff
@@ -393,14 +394,20 @@ class _CarouselViewState extends State<CarouselView> {
   }
 
   Widget _buildCarouselItem(ThemeData theme, int index) {
-    final EdgeInsets effectivePadding = widget.padding ?? const EdgeInsets.all(4.0);
+    final CarouselViewThemeData carouselTheme = theme.carouselViewTheme;
+    final EdgeInsets effectivePadding =
+        widget.padding ?? carouselTheme.padding ?? const EdgeInsets.all(4.0);
     final Color effectiveBackgroundColor =
-        widget.backgroundColor ?? Theme.of(context).colorScheme.surface;
-    final double effectiveElevation = widget.elevation ?? 0.0;
+        widget.backgroundColor ??
+        carouselTheme.backgroundColor ??
+        Theme.of(context).colorScheme.surface;
+    final double effectiveElevation = widget.elevation ?? carouselTheme.elevation ?? 0.0;
     final ShapeBorder effectiveShape =
         widget.shape ??
+        carouselTheme.shape ??
         const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(28.0)));
     final WidgetStateProperty<Color?> effectiveOverlayColor =
+        widget.overlayColor ??
```

Comment:
This line is duplicated? You probably mean to use the CarouselTheme's overlay color.